crux-llvm/src/Crux/LLVM/Simulate.hs

```diff
@@ -125,55 +125,78 @@ registerFunctions llvm_module mtrans =
-
-                 let explainFailure evalFn gl =
-                       do bb <- readIORef bbMapRef
-                          ex <- explainCex sym bb evalFn >>= \f -> f (gl ^. labeledPred)
-                          let details = case ex of
-                                NoExplanation -> mempty
-                                DisjOfFailures xs ->
-                                  case map ppBB xs of
-                                    []  -> mempty
-                                    [x] -> indent 2 x
-                                    xs' | length xs' <= detailLimit
-                                        -> "All of the following conditions failed:" <> line <> indent 2 (vcat xs')
-                                        | otherwise
-                                        -> "All of the following conditions failed (and other conditions have been elided to reduce output): "
-                                               <> line <> indent 2 (vcat (take detailLimit xs'))
-
-                          return $ vcat [ ppSimError (gl^.labeledPredMsg), details ]
-
-                 return (Crux.RunnableState initSt, explainFailure)
+simulateLLVMFile llvm_file llvmOpts = do
+  Crux.SimulatorCallback $ \sym maybeOnline -> do
+    halloc <- newHandleAllocator
+    bbMapRef <- newIORef (Map.empty :: LLVMAnnMap sym)
+    let ?recordLLVMAnnotation = \an bb -> modifyIORef bbMapRef (Map.insert an bb)
+    runnableState <- setupFileSim halloc llvm_file llvmOpts sym maybeOnline
+    return (runnableState, explainFailure sym bbMapRef)
+
+setupFileSim :: Logs
+             => IsSymInterface sym
+             => sym ~ WEB.ExprBuilder t st fs
+             => HasLLVMAnn sym
+             => HandleAllocator
+             -> FilePath
+             -> LLVMOptions
+             -> sym -> Maybe (Crux.SomeOnlineSolver sym)
+             -> IO (Crux.RunnableState sym)
+setupFileSim halloc llvm_file llvmOpts sym _maybeOnline =
+  do memVar <- mkMemVar "crux:llvm_memory" halloc
+
+     let simctx = (setupSimCtxt halloc sym (memOpts llvmOpts) memVar)
+                  { printHandle = view outputHandle ?outputConfig }
+
+     prepped <- prepLLVMModule llvmOpts halloc sym llvm_file memVar
+
+     let globSt = llvmGlobals memVar (p_mem prepped)
+
+     return $ Crux.RunnableState $
+       InitialState simctx globSt defaultAbortHandler UnitRepr $
+       runOverrideSim UnitRepr $
+       do Some trans <- return $ p_someTrans prepped
+          llvmPtrWidth (trans ^. transContext) $ \ptrW ->
+            withPtrWidth ptrW $
+            do registerFunctions (p_llvmMod prepped) trans
+               checkFun (entryPoint llvmOpts) (cfgMap trans)
+
+
+
+
+data PreppedLLVM sym = PreppedLLVM { p_llvmMod :: LLVM.Module
+                                   , p_someTrans :: Some ModuleTranslation
+                                   , p_memVar :: GlobalVar Mem
+                                   , p_mem :: MemImpl sym
+                                   }
```

Comment:
The majority of record accessors in Crucible use `camelCase`, I'd recommend keeping that scheme as consistent as possible and removing the underscores in the accessors here.

Reply:
Thanks!  I had initially used  underscores as a signal to myself that this was a tentative change and missed cleaning these up when I decided to go forward on this implementation.